src/App.tsx

```diff
@@ -1,26 +1,207 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { getTodos, addTodo, deleteTodo } from './api/todos';
+import { USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import Header from './components/Header';
+import TodoList from './components/TodoList';
+import Footer from './components/Footer';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const [filter, setFilter] = useState('all');
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  // Реф для фокусування на полі вводу
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      setLoading(true);
+      setError(null);
+
+      try {
+        const todosFromApi = await getTodos();
+
+        setTodos(todosFromApi);
+      } catch (apiError) {
+        setError('Unable to load todos');
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    if (USER_ID) {
+      loadTodos();
+    }
+  }, []);
+
+  useEffect(() => {
+    if (error) {
+      const timer = setTimeout(() => {
+        setError(null);
+      }, 3000);
+
+      return () => clearTimeout(timer);
+    }
+  }, [error]);
+
+  const filteredTodos = todos.filter(todo => {
+    if (filter === 'active') {
+      return !todo.completed;
+    }
+
+    if (filter === 'completed') {
+      return todo.completed;
+    }
+
+    return true;
+  });
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!newTodoTitle.trim()) {
+      setError('Title should not be empty');
+
+      return;
+    }
+
+    const trimmedTitle = newTodoTitle.trim();
+
+    setIsSubmitting(true);
+
+    const newTempTodo: Todo = {
+      id: Date.now(),
+      userId: USER_ID,
+      title: trimmedTitle,
+      completed: false,
+    };
+
+    setTempTodo(newTempTodo);
+    setTodos(prevTodos => [...prevTodos, newTempTodo]);
+
+    try {
+      const createdTodo = await addTodo({
+        userId: USER_ID,
+        title: trimmedTitle,
+        completed: false,
+      });
+
+      setTodos(prevTodos =>
+        prevTodos.map(todo =>
+          todo.id === newTempTodo.id ? createdTodo : todo,
+        ),
+      );
+
+      setNewTodoTitle('');
+
+      setTimeout(() => {
+        if (inputRef.current) {
+          inputRef.current.focus();
+        }
+      }, 0);
+    } catch {
+      setError('Unable to add a todo');
+      setTodos(prevTodos =>
```

Comment:
You shouldn't add a tempTodo to array of todos
You display temp todo separately in TodoList